eip/EIPS/eip-4337.md

```diff
@@ -624,6 +624,219 @@ Returns an array of the entryPoint addresses supported by the client. The first
 }
 ```
 
+#### * eth_chainId
+
+Returns EIP155 Chain ID.
+
+```json=
+# Request
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "method": "eth_chainId",
+  "params": []
+}
+
+# Response
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": "0x1"
+}
+```
+
+### debug Namespace
+
+#### * debug_bundler_clearState
+
+Clears the bundler mempool and reputation data of paymasters/accounts/factories/aggregators.
+
+```json=
+# Request
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "method": "debug_bundler_clearState",
+  "params": []
+}
+
+# Response
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": "ok"
+}
+```
+
+#### * debug_bundler_dumpMempool
+
+Returns:
+
+`array` - Array of mempool entries objects, or an empty array.
+
+where each mempool entry has the following fields:
+
+`userOp` - UserOperation object. See `eth_sendUserOperation` for more details.
+`prefund` - The calculated prefund required for the user operation execution.
+`aggregator` - (optional) the aggregator used by the senderaccount.
+
+```json=
+# Request
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "method": "debug_bundler_dumpMempool",
+  "params": []
+}
+
+# Response
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": [
+    {
+      "userOp": {
+        sender, // address
+        nonce, // uint256
+        initCode, // bytes
+        callData, // bytes
+        callGasLimit, // uint256
+        verificationGasLimit, // uint256
+        preVerificationGas, // uint256
+        maxFeePerGas, // uint256
+        maxPriorityFeePerGas, // uint256
+        paymasterAndData, // bytes
+        signature // bytes
+      },
+      "prefund": "0x1234abcd",
+      "aggregator": "0x7A0A0d159218E6a2f407B99173A2b12A6DDfC2a6"
+    }
+  ]
+}
+```
+
+#### * debug_bundler_sendBundleNow
+
+Forces the bundler to execute the entire current mempool as `handleOps()` transaction.
```

Comment:
It's not necessarily the entire bundle, as it may be impossible due to conflicts between ops, paymaster having insufficient funds for all ops, etc.  I think it triggers building and sending the maximum bundler possible.  That is, the most profitable bundle that can be built from the current mempool.  E.g. if there are two conflicting ops, the one with the highest priority fee will be simulated first and therefore included instead of the other one.

Some tests may actually rely on this behavior, and dump the mempool before and after, to see which ones were left behind.

Reply:
yes. this is more of "trigger the logic to build the next bundle from the mempool" rather than "force"

Reply:
This behavior is only expected on test mode, so I assumed the bundler already dropped conflicting ops. So it shouldn't be in the mempool at that point anyway, no?